File: Backend/src/models/Product.ts
import { UserDoc } from "./User.js";
import { model, Schema } from "mongoose";

export type ProductDoc = {
  _id: string;
  name: string;
  desc: string;
  category: string;
  rating: number;
  thumbnail: string;
  images: String[];
  price: number;
  discount: number;
  availability: boolean;
  sellerId: Schema.Types.ObjectId | UserDoc;
  dealId: Schema.Types.ObjectId;
  stockUnit: number;
};
const ProductSchema = new Schema(
  {
    name: String,
    desc: String,
    category: String,
    rating: Number,
    thumbnail: String,
    images: [String],
    price: Number,
    discount: Number,
    brand: String,
    availability: {
      type: Boolean,
      default: true,
    },
    sellerId: {
      type: Schema.Types.ObjectId,
      ref: "User",
    },
    dealId: Schema.Types.ObjectId,
    stockUnit: Number,
  },
  { timestamps: true }
);

export const Product = model("Product", ProductSchema);

File: python/Backend/application/models/Product.py
from application.models.db import *
from application.models.User import User
class Product(Document):
    name = StringField()
    desc = StringField()
    category= StringField()
    rating = FloatField()
    thumbnail = StringField()
    images= ListField(StringField())
    price = IntField()
    discount = FloatField()
    brand = StringField()
    availability = BooleanField(default=True)
    sellerId = ReferenceField(User, reverse_delete_rule=CASCADE)
    stockUnit= IntField()

    
    def to_json(self):
        return {
            "name": self.name,
            "desc": self.desc,
            "category": self.category,
            "rating": self.rating,
            "thumbnail": self.thumbnail,
            "images": self.images,
            "price": self.price,
            "discount": self.discount,
            "brand": self.brand,
            "availability": self.availability,
            "sellerId": self.sellerId,
            "dealId": self.dealId,
            "stockUnit": self.stockUnit,
        }
